### PR TITLE
Adds click events and websocket parameter

### PR DESCRIPTION
**Pull request by farisca**

### CODE REVIEW

File: js/pokemap.js [hunk 1]
```diff
@@ -4,37 +4,39 @@ var util = require('util');
...
-  apiURL = options.apiEndPoint;
-
+  apiEndpoint = options.apiEndpoint;
+  websocketEndpoint = options.websocketEndpoint;
 
   // which pokemons should be shown; if null show all pokemons; otherwise only pokemons with ids in the list
   this.filterPokemons = null;
 
-  //this.socket = io.connect('http://localhost:3000');
+  socket = io.connect(websocketEndpoint + '/mobs');
```
Comment (johartl) [inline js/pokemap.js:28]: - [ ] As far as I understand you only need to do `io.connect(websocketEndpoint)` here. The `/mobs` part will become important later when you do `socket.on('mobs', ..)`.

File: js/pokemap.js [hunk 2]
```diff
@@ -4,37 +4,39 @@ var util = require('util');
 var L = require('leaflet');
 var io = require('socket.io-client');
 var socket = io.connect("http://localhost:3000");
```
Comment (johartl) [inline js/pokemap.js]: This should be removed.

**Comment by johartl** [comment]
- [ ] Please also remove the `var socket = io.connect("http://localhost:3000");` statement at the top.

**Comment by johartl** [comment]
- [ ] The way you form the URLs for the API calls are wrong. You need to prepend `/api` before every URL.

```
// wrong
var URL=  apiEndpoint + "/pokemon/sighting";

// correct
var URL=  apiEndpoint + "/api/pokemon/sighting";
```

**Comment by johartl** [comment]
- [ ] The dependencies for `leaflet-routing-machine` and `leaflet-control-geocoder` are missing in your package.json

**Comment by johartl** [comment]
@farisca Are you on this?

**Comment by timobrembeck** [comment]
outdated branch

**Pull request closed without merging**